yt_dlp/extractor/yandexdisk.py

```diff
@@ -62,6 +66,39 @@ def _real_extract(self, url):
             webpage, 'store'), video_id)
         resource = store['resources'][store['rootResourceId']]
 
+        if store['rootResourceId'] == 'password-protected':
+            data = {
+                'hash': resource['hash'],
+                'password': self.get_param('videopassword', default=''),
+                'sk': traverse_obj(store, ('environment', 'sk'))}
+            json_string = json.dumps(data, separators=(',', ':'))
+            url_encoded_string = urllib.parse.quote(json_string, safe='')
+            data_bytes = url_encoded_string.encode('utf-8')
+            token = (self._download_json('https://disk.yandex.ru/public/api/check-password',
+                video_id, data=data_bytes, fatal=False,
+                headers={
+                    'Accept': '*/*',
+                    'Cache-Control': 'no-cache',
+                    'Connection': 'keep-alive',
+                    'Content-Type': 'text/plain',
+                    'Origin': 'https://disk.yandex.ru',
+                    'Pragma': 'no-cache',
+                    'Referer': url,
+                    'Sec-Fetch-Dest': 'empty',
+                    'Sec-Fetch-Mode': 'cors',
+                    'Sec-Fetch-Site': 'same-origin',
+                    'X-Requested-With': 'XMLHttpRequest',
+                    'X-Retpath-Y': url}) or {}).get('token') or {}
+            if not token:
+                raise ExtractorError('Password incorrect!', expected=True)
+            self._set_cookie('disk.yandex.ru', name='passToken', value=token)
+            webpage = self._download_webpage(url, video_id)
+            store = self._parse_json(self._search_regex(
+                r'<script[^>]+id="store-prefetch"[^>]*>\s*({.+?})\s*</script>',
+                webpage, 'store'), video_id)
+            resource = store['resources'][store['rootResourceId']]
+
+        thumbnail = unescape(self._og_search_property('image', webpage))
```

Comment:
`unescape` is not needed here; the value returned by `_og_search_property()` has been run through `yt_dlp.utils.unescapeHTML` (which should be used instead of `html.unescape`